### DISCUSSION TITLE
Authentication with OAuth2 : Client token expired

### DISCUSSION THREAD

**Discussion by navaneeth-raman-bhaskar** [spatie/laravel-google-calendar#224] [Q&A]
After setting `GOOGLE_CALENDAR_AUTH_PROFILE=oauth` in `.env`  Application worked.
After some days `oauth-token.json`  is expired .
How to refresh this token using this package automatically?

**Comment by EmilioPuljiz** [comment]
I have the same problem

**Comment by tomcodes** [comment]
Hi there, late to the party but this package is not made to renew your tokens, it's something you have to manage by yourself.

Personally, I have a cron setup that renews the tokens everyday so that even users that don't log in my app very often always have a working synd with their Google Calendar.

I use League OAuth-client to renew my access tokens (be careful, version 2.8.0 introduced a regression with the scope management).

See `getAccessToken` in this doc: https://github.com/thephpleague/oauth2-client/blob/master/docs/usage.md
